generators/javascript/logic.ts

```diff
@@ -10,10 +10,12 @@
 
 // Former goog.module ID: Blockly.JavaScript.logic
 
+import type {Block} from '../../core/block.js';
+import type {JavascriptGenerator} from './javascript_generator.js';
 import {Order} from './javascript_generator.js';
 
 
-export function controls_if(block, generator) {
+export function controls_if(block: Block, generator: JavascriptGenerator) {
```

Comment:
Expression blocks require an explicit return type of `[string, Order]` as the return type is otherwise inferred to be `(string | Order)[]`, which causes all kinds of type errors in the places where the return values are consumed.

For statements, the return type is correctly inferred from the `return` statement (either a `string` or `null`).  Do you think it worthwhile to specify the return type explicitly?  If so, do you think it preferable to specify it as _either_ `string` or `null`, or as `string | null`?

Or, should they _all_ be `string | null | [string, Order]`?